HeatEquationSolver/heat_equation.py

```diff
@@ -20,28 +30,82 @@ def initialise_grid(length, nodes, time, k):
     return dx, dt, time_step, x, t
 
 
-def setting_conditions(t, time_step, nodes):
-    # initial and boundary conditions
-    u_initial = 25
-    u_left = 20 * np.sin(np.pi * t) + 25
-    u_right = t + 25
+def left_temp_func(temp, t):
+    """
+    Generates the temperature at the left boundary of the rod
+
+    :param temp: initial temperature of the rod at t=0
+    :param t: array consisting of time steps from t=0 to t= total time
+    :return: the temperature at the left boundary
+    """
+    return 20 * np.sin(np.pi * t) + temp
+
+
+def right_temp_func(temp, t):
+    """
+    Generates the temperature at the right boundary of the rod
+
+    :param temp: initial temperature of the rod at t=0
+    :param t: array consisting of time steps from t=0 to t= total time
+    :return: the temperature at the right boundary
+    """
+    return t + temp
+
+
+def setting_conditions(t, time_step, nodes, temp):
+    """
+    Initialises the temperature distribution and boundary conditions for the heat equation
+
+    :param t: array consisting of time steps from t=0 to t=total time
+    :param time_step: no of time steps for the simulation
+    :param nodes: no of spatial nodes
+    :param temp: initial temperature of the rod at t=0
+    :return:    - u: 2d-matrix initialised with the initial temperature of the rod
+                - u_left: array denoting the temperature at the left boundary over time
+                - u_right: array denoting the temperature at the right boundary over time
+    """
+    u_initial = temp
+    u_left = left_temp_func(temp, t)
+    u_right = right_temp_func(temp, t)
     u = np.zeros((time_step, nodes))
     u[0, :] = u_initial
 
     return u, u_left, u_right
 
 
-# initializing 2d-matrix to store temperature
 def solve_heat_equation(u, u_left, u_right, time_step, nodes, dt, k, dx):
+    """
+    Solves the heat equation using the explicit scheme
+
+    :param u: 2d-matrix initialised with the initial temperature of the rod
+    :param u_left: array denoting the temperature at the left boundary over time
```

Comment:
Are `u_left` and `u_right` arrays or functions?

Reply:
u_left and u_right are arrays consisting of varying temperatures